src/MovieInfo/MovieInfo.js

```diff
@@ -1,24 +1,76 @@
-import React from 'react'
+import React, { Component } from 'react'
 import './MovieInfo.css'
 
-const MovieInfo = (props) => {
-  const movie = props.movie;
+class MovieInfo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movie: props.movie,
+      rated: false,
+      userRating: 0,
+      isCurrentUser: props.isCurrentUser
+    }
+  }
+  // const movie = props.movie;
 
-  return (
-    <article className="movie-info">
-      <img src={movie.backdrop_path} alt={`${movie.title} backdrop`}></img>
-      <h2>{movie.title}</h2>
-      <p data-testid='release date'>Release Date: {movie.release_date}</p>
-      <p data-testid='overview'>Overview: {movie.overview}</p>
-      <ul data-testid='genres'>Genres: {movie.genres.map((genre, index) => {
-        return <li key={index}>{genre}</li>
-      })}</ul>
-      <p data-testid='budget'>Budget: {movie.budget}</p>
-      <p data-testid='runtime'>Runtime: {movie.runtime} days</p>
-      <p data-testid='tagline'>Tagline: {movie.tagline}</p>
-      <p data-testid='average rating'>Average Rating: {movie.average_rating}</p>
-    </article>
-  )
+  preventDefault = (e) => {
+    e.preventDefault()
+  }
+
+  resetInput = (e) => {
+    e.target.value = ''
+  }
+
+  createRating = (e) => {
+    this.setState({userRating: parseInt(e.target.value)})
+  }
+
+  rateMovie = (props) => {
+    if(this.state.userRating > 0 && this.state.isCurrentUser) {
+      console.log(this.state.rated)
+      this.props.submitRating(this.state.userRating)
+    } else {
+      console.log('whoops')
+    }
+  }
+  render() {
+    return (
+      <article className="movie-info">
+        <img src={this.state.movie.backdrop_path} alt={`${this.state.movie.title} backdrop`}></img>
+        <h2>{this.state.movie.title}</h2>
+        <p data-testid='release date'>Release Date: {this.state.movie.release_date}</p>
+        <p data-testid='overview'>Overview: {this.state.movie.overview}</p>
+        <ul data-testid='genres'>Genres: {this.state.movie.genres.map((genre, index) => {
+          return <li key={index}>{genre}</li>
+        })}</ul>
+        <p data-testid='budget'>Budget: {this.state.movie.budget}</p>
+        <p data-testid='runtime'>Runtime: {this.state.movie.runtime} days</p>
+        <p data-testid='tagline'>Tagline: {this.state.movie.tagline}</p>
+        <p data-testid='average rating'>Average Rating: {this.state.movie.average_rating.toFixed(1)}</p>
```

Comment:
Now that we're targeting these in a different way, we could get rid of these test-ids